galasa-parent/dev.galasa.framework.api.ras/src/test/java/dev/galasa/framework/api/ras/internal/TestRunQuery.java

```diff
@@ -0,0 +1,656 @@
+			this.resp = new MockHttpServletResponse(this.writer);
+			this.servlet = new RunQuery();
+			this.servlet.framework = (IFramework) this.mockFramework;
+		}
+
+		public HttpServletResponse getResponse (){
+			return this.resp;
+		}
+
+		public HttpServletRequest getRequest (){
+			return this.req;
+		}
+
+		public RunQuery getServlet(){
+			return this.servlet;
+		}
+
+		public ByteArrayOutputStream getOutStream(){
+			return this.outStream;
+		}
+
+		public void setMockInputs(List<IRunResult> mockInpResults){
+			this.mockInputRunResults = mockInpResults;
+		}
+
+		public List<IResultArchiveStoreDirectoryService> getDirectoryService(MockResultArchiveStoreDirectoryService rasStore){
+			List<IResultArchiveStoreDirectoryService> directoryServices = new ArrayList<IResultArchiveStoreDirectoryService>();
+			directoryServices.add(rasStore);
+			return directoryServices;
+		}
+
+		public void setArchiveStore(MockArchiveStore store){
+			this.archiveStore = store;
+		}
+
+	}
+
+	public List<IRunResult> generateTestData (int resSize, int passTests){
+		List<IRunResult> mockInputRunResults = new ArrayList<IRunResult>();
+		int passCount = 0;
+		// Build the results the DB will return.
+		for(int c =0 ; c < resSize; c++){
+			String runName = RandomStringUtils.randomAlphanumeric(5);
+			String requestor = RandomStringUtils.randomAlphanumeric(8);
+			String runId = RandomStringUtils.randomAlphanumeric(16);
+			TestStructure testStructure = new TestStructure();
+			testStructure.setRunName(runName);
+			testStructure.setRequestor(requestor);
+			if (passCount < passTests){
+				testStructure.setResult("Passed");
+				passCount ++;
+			}else{
+				testStructure.setResult("Failed");
+			}
+			Path artifactRoot = Paths.get(RandomStringUtils.randomAlphanumeric(12));
+			String log = RandomStringUtils.randomAlphanumeric(6);
+			IRunResult result = new MockRunResult( runId, testStructure, artifactRoot , log);
+			mockInputRunResults.add(result);
+		}
+		return mockInputRunResults;
+	}
+
+	public String generateExpectedJson (List<IRunResult> mockInputRunResults, String[] pageSize,String[] pageNo){
+		int resSize = Integer.parseInt(pageSize[0]);
+		double numPages = Math.ceil((double)mockInputRunResults.size()/resSize);
+		if (numPages == 0) {
+			numPages = 1;
+		}
+		String jsonResult = "{\n"+
+							"  \"pageNum\": "+pageNo[0]+",\n"+
+							"  \"pageSize\": "+pageSize[0]+",\n"+
+							"  \"numPages\": "+(int)numPages+",\n"+
+							"  \"amountOfRuns\": "+mockInputRunResults.size()+",\n"+
+							"  \"runs\": [";
+		if (mockInputRunResults.size()>0){	
+			int iter;
+			if (resSize < mockInputRunResults.size() ){
+				iter = resSize;
+			}else{
+				iter= mockInputRunResults.size();
+			}
+			int pagedResult = (Integer.parseInt(pageNo[0])-1)*resSize;
+			for ( int c= 0; c< iter; c++ ){
+				String runData ="";
+				if (0<c && c<iter){
+					runData =",\n";
+				}else if(c==0){
+					runData ="\n";
+				}
+				try {
+					runData = runData+ "    {\n"+  
```

Comment:
Could have saved a ton of effort by removing all the whitespace from the actual and expected strings within the assert statements. Then formatting would have been removed as an issue, and the tests would have still worked were it to change to passing min/compressed-style formatting.

Reply:
Thank you. Will definitely bear in mind for future use.